spec/mongo/cluster_spec.rb

```diff
@@ -20,6 +20,24 @@
 
   let(:cluster) { cluster_without_io }
 
+  describe 'initialize' do
+
+    context 'when there are duplicate addresses' do
+
+      let(:addresses) do
+        SpecConfig.instance.addresses + SpecConfig.instance.addresses
+      end
+      let(:cluster_with_dup_addresses) do
+        described_class.new(addresses, monitoring, SpecConfig.instance.test_options)
```

Comment:
I believe your tests are failing because you're allowing monitoring_io on this cluster. Should be fixed if you merge `monitoring_io: false` with `SpecConfig.instance.test_options`.

Reply:
Hi @egiurleo thank you for the rapid feedback. Which tests are failing? Are they ones outside of this file? When I run the spec file locally I get a 100% pass rate.

Unfortunately if I merge in `monitoring_io: false` into the test options, the test won't fail in the first place without the patch.

Reply:
Yeah, the tests are outside of this file because background threads are still running even after all the tests have terminated. Of course, as you said, the test will be pointless with `monitoring_io: false` -- thanks for pointing that out.

Could you try disconnecting the cluster you create after the test is run? Something like:

```
after do
  cluster_with_dup_addresses.disconnect!
```

I'll also defer to @p-mongo on whether that's the right solution.

Reply:
You got it, added.

Reply:
Thank you! I'll authorize the test run.

Reply:
The way this should be handled is with `register_cluster` - see e.g. https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/mongo/cluster_spec.rb#L10.

Reply:
I updated to do that.